### PR TITLE
James 1929 JMAP wiring for hasAttachment

### PR DESCRIPTION
**Pull request by chibenwa**
Only the commits marked as JAMES-1929 (two last commits)

### CODE REVIEW

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/impl/PropertyBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+public class PropertyBuilderTest {
+
+    @Test
+    public void emptyPropertyBuilderShouldCreateEmptyProperties() {
+        assertThat(new PropertyBuilder().toProperties()).isEmpty();
+    }
+
+    @Test
+    public void setHasAttachmentShouldAddTrueWhenCalledWithTrue() {
```
Comment (quynhn) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/impl/PropertyBuilderTest.java]: It seems that having naming function and assigned value. Name with "true" but I setHasAttachment(false).
Is that correct purpose?

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/impl/PropertyBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        PropertyBuilder propertyBuilder = new PropertyBuilder();
+        propertyBuilder.setHasAttachment(false);
+        assertThat(propertyBuilder.toProperties())
+            .containsOnly(new SimpleProperty(PropertyBuilder.JAMES_INTERNALS, PropertyBuilder.HAS_ATTACHMENT, "false"));
+    }
+
+    @Test
+    public void setHasAttachmentShouldAddFalseWhenCalledWithFalse() {
```
Comment (quynhn) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/impl/PropertyBuilderTest.java]: And here?

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java [hunk 3]
```diff
@@ -129,6 +129,52 @@ public void getMessageListShouldReturnErrorInvalidArgumentsWhenHeaderIsInvalid()
...
+            .body("[[\"getMessageList\", {\"filter\":{\"hasAttachment\":\"false\"}}, \"#0\"]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("messageList"))
+            .body(ARGUMENTS + ".messageIds", contains(message1.getMessageId().serialize(), message3.getMessageId().serialize()));
+    }
+
+    @Test
```
Comment (quynhn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: Do we need to care about testing with email which has mixing attachment and inlined?

Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: No, such detailed testing can be handled at the mailbox layer

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java (hunk 3 above)
Comment (quynhn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: :+1:

File: mailbox/api/src/main/java/org/apache/james/mailbox/model/SearchQuery.java [hunk 4]
```diff
@@ -595,12 +595,20 @@ public static Criterion flagSet(Flag flag, boolean isSet) {
         return result;
     }
 
+    public static Criterion hasAttachment(boolean value) {
+        if (value) {
+            return new AttachmentCriterion(BooleanOperator.set());
+        } else {
```
Comment (quynhn) [inline mailbox/api/src/main/java/org/apache/james/mailbox/model/SearchQuery.java]: Don't we need "else" or just return ...

Comment (chibenwa) [inline mailbox/api/src/main/java/org/apache/james/mailbox/model/SearchQuery.java]: We can...

**Comment by chibenwa** [comment]
Integrated to https://github.com/linagora/james-project/pull/619

**Pull request closed without merging**